examples/client.rs

```diff
@@ -72,13 +72,13 @@ fn main() {
             println!("padded audio file with {} characters", padding);
         }
     }
-    let buffers = audio_buf[..].chunks(FRAME_SIZE);
     let mut denoised_buffer: Vec<f32> = vec![];
     let mut rnnoise = DenoiseState::new();
+    let mut denoised_chunk: Vec<f32> = vec![0.0; FRAME_SIZE];
+    let buffers = audio_buf[..].chunks(FRAME_SIZE);
     for buffer in buffers {
-        let mut denoised: Vec<f32> = vec![0.0; FRAME_SIZE];
-        rnnoise.process_frame_mut(&buffer, &mut denoised[..]);
-        denoised_buffer.append(&mut denoised);
+        rnnoise.process_frame_mut(&buffer, &mut denoised_chunk[..]);
+        denoised_buffer.append(&mut denoised_chunk.clone());
```

Comment:
The clone shouldn't be needed.

Reply:
Actually you should be using `extend_from_slice` here, not append, because f32 is cheap to copy.

Reply:
With the clone it works correctly:
```
 cargo run --example client -- -v -i test.wav -o output.raw
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/examples/client -v -i test.wav -o output.raw`
Configuration: Configuration { verbose: true, input: "test.wav", output: "output.raw" }
test.wav length: 51273
padded audio file with 393 characters
output.raw length: 51360
```

Without the clone `denoised_chunk` becomes zero-length and it fails:
```
$ cargo run --example client -- -v -i test.wav -o output.raw
   Compiling rnnoise-c v0.1.0 (/home/jandrews/devel/rust/rnnoise-c)
    Finished dev [unoptimized + debuginfo] target(s) in 2.41s
     Running `target/debug/examples/client -v -i test.wav -o output.raw`
Configuration: Configuration { verbose: true, input: "test.wav", output: "output.raw" }
test.wav length: 51273
padded audio file with 393 characters
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `480`', src/lib.rs:22:3
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Reply:
ah, I learned something new, thanks! That works, no clone